02B-Python-Omop-walkthrough-critical-care/02B-Python-Omop-walkthrough-critical-care.md

```diff
@@ -0,0 +1,2006 @@
+This document is an introductory walkthrough, demonstrating how to read into Python some OMOP data from a series of csv files, join on concept names and explore and visualise the data.
+
+It uses synthetic data created in UCLH as a part of the  [Critical Care Health Informatics Collaborative (CCHIC) project](https://safehr-data.github.io/uclh-research-discovery/projects/uclh_cchic_s0/index.html).
+
+# Installing & loading required packages
+1. ### pip install pandas
+
+2. ### pip install seaborn
+
+3. ### pip install matplotlib
+
+
+
+```python
+import os
+import requests
+import pandas as pd
+import datetime
+import seaborn as sns
+import matplotlib.pyplot as plt
+```
+
+# Downloading & reading in OMOP data
+Here we will download & read in some UCLH critical care data stored in Github
+
+
+```python
+
+
+# Repository and file paths
+repository = "SAFEHR-data/uclh-research-discovery"
+source_path = "_projects/uclh_cchic_s0/data"
+destination_folder = "Starter_Guide_python/data"
+
+# Create destination folder if it doesn't exist
+os.makedirs(destination_folder, exist_ok=True)
+
+# GitHub API URL to get the contents of the folder
+api_url = f"https://api.github.com/repos/{repository}/contents/{source_path}"
+
+# Fetch the list of files from the GitHub folder
+files = requests.get(api_url).json()
+
+# Download each file
+for file in files:
+    file_url = file["download_url"]
+    file_name = file["name"]
+    file_path = os.path.join(destination_folder, file_name)
+
+    # Download the file and save it locally
+    print(f"Downloading {file_name} ******")
+    file_response = requests.get(file_url)
+
+    # Save the file to the local folder
+    with open(file_path, "wb") as local_file:
+        local_file.write(file_response.content)
+
+# List all downloaded files
+print("Download complete! Files in the folder:")
+print(os.listdir(destination_folder))
+
+```
+
+    Downloading condition_occurrence.csv ******
+    Downloading death.csv ******
+    Downloading device_exposure.csv ******
+    Downloading drug_exposure.csv ******
+    Downloading measurement.csv ******
+    Downloading observation.csv ******
+    Downloading observation_period.csv ******
+    Downloading person.csv ******
+    Downloading procedure_occurrence.csv ******
+    Downloading specimen.csv ******
+    Downloading visit_occurrence.csv ******
+    Download complete! Files in the folder:
+    ['observation_period.csv', 'drug_exposure.csv', 'specimen.csv', 'death.csv', 'device_exposure.csv', 'measurement.csv', 'condition_occurrence.csv', 'visit_occurrence.csv', 'person.csv', 'observation.csv', 'procedure_occurrence.csv']
+
+
+# Loading Multiple CSV Files into DataFrames
+This script loads all CSV files from a specified folder into individual pandas DataFrames and stores them in a dictionary.
+
+
+```python
+#########Load files in Data frames w.r.t file names
+# store the loaded data from each CSV file.
+df = {} 
```

Comment:
I'd suggest this is called `cdm` so that it follows with the R code. If you don't like that then something like `dataframes` so that it shows that is multiple of them